pkg/operator/resources/operator/operator.go

```diff
@@ -0,0 +1,598 @@
+									Type: "array",
+								},
+							},
+						},
+						"status": {
+							Type: "object",
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+// NewClusterServiceVersion creates all cluster resources fr a specific group/component
+func NewClusterServiceVersion(data *ClusterServiceVersionData) (*csvv1.ClusterServiceVersion, error) {
+	deployment := createOperatorDeployment(
+		operatorName,
+		data.Namespace,
+		data.OperatorImage,
+		data.ImagePullPolicy,
+		"name",
+		operatorName,
+		int32(1),
+	)
+
+	strategySpec := csvStrategySpec{
+		ClusterPermissions: []csvPermissions{
+			{
+				ServiceAccountName: serviceAccountName,
+				Rules:              getClusterPolicyRules(),
+			},
+		},
+		Deployments: []csvDeployments{
+			{
+				Name: operatorName,
+				Spec: deployment.Spec,
+			},
+		},
+	}
+
+	strategySpecJSONBytes, err := json.Marshal(strategySpec)
+	if err != nil {
+		return nil, err
+	}
+
+	csvVersion, err := semver.New(data.CsvVersion)
+	if err != nil {
+		return nil, err
+	}
+
+	return &csvv1.ClusterServiceVersion{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ClusterServiceVersion",
+			APIVersion: "operators.coreos.com/v1alpha1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      operatorName + "." + data.CsvVersion,
+			Namespace: data.Namespace,
+		},
+		Spec: csvv1.ClusterServiceVersionSpec{
+			DisplayName: "VM import operator",
+			Description: "VM import operator provides ability to import virtual machines from other infrastructure like oVirt/RHV",
+			Keywords:    []string{"Import", "Virtualization", "oVirt", "RHV"},
+			Version:     version.OperatorVersion{Version: *csvVersion},
+			Maturity:    "alpha",
+			Replaces:    data.ReplacesCsvVersion,
+			Maintainers: []csvv1.Maintainer{{
+				Name:  "KubeVirt project",
+				Email: "kubevirt-dev@googlegroups.com",
+			}},
+			Provider: csvv1.AppLink{
+				Name: "KubeVirt project",
+			},
+			Links: []csvv1.AppLink{
+				{
+					Name: "VM import operator",
+					URL:  "https://github.com/kubevirt/vm-import-operator/blob/master/README.md",
+				},
+				{
+					Name: "Source Code",
+					URL:  "https://github.com/kubevirt/vm-import-operator/",
+				},
+			},
+			Labels: map[string]string{
+				"operated-by": operatorName,
+			},
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"operated-by": operatorName,
+				},
+			},
+			InstallModes: []csvv1.InstallMode{
+				{
+					Type:      csvv1.InstallModeTypeOwnNamespace,
+					Supported: true,
+				},
+				{
+					Type:      csvv1.InstallModeTypeSingleNamespace,
+					Supported: true,
+				},
+				{
+					Type:      csvv1.InstallModeTypeAllNamespaces,
+					Supported: true,
+				},
+			},
+			InstallStrategy: csvv1.NamedInstallStrategy{
+				StrategyName:    "deployment",
+				StrategySpecRaw: json.RawMessage(strategySpecJSONBytes),
+			},
+			CustomResourceDefinitions: csvv1.CustomResourceDefinitions{
+				// TODO: should resource mappings crd be included here as well or only watched CRs ?
```

Comment:
Is this TODO still valid?

Reply:
I'll remove the comment. we'll have to add generation of CRDs by code but they will be added differently and not in this context.